ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -95,6 +97,21 @@ public IncrementalLoadTasksBuilder(String dbName, String tableName, String loadP
     this.log.debug("Iteration num " + numIteration);
     TaskTracker tracker = new TaskTracker(conf.getIntVar(HiveConf.ConfVars.REPL_APPROX_MAX_LOAD_TASKS));
 
+    // if no events are there to replay, then update the last repl id of the database/ table to last event id.
+    if (!iterator.hasNext()) {
+      DumpMetaData eventDmd = new DumpMetaData(new Path(loadPath), conf);
```

Comment:
Can we set eventTo in ReplLoadWork from ReplicationSemanticAnalyzer as we already read DumpMetaData there and use it here?

Reply:
yes ..can be done ..but i dont see much improvement as this flow is not normal flow